webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -223,25 +223,26 @@ public void pull(Repository repository,
                      List<String> optionList) {
 
         SmartlingOptions options = SmartlingOptions.parseList(optionList);
-        AndroidStringDocumentMapper mapper = new AndroidStringDocumentMapper(pluralSeparator, null);
```

Comment:
can matching ctor be removed?

Reply:
We can't, since its being used for other use cases.

Reply:
not sure in what case we don't have the repository name and locale name, even i'm confused why it is fine not having it. in fact relooking at the class I'm not sure why that class is not a stateless component?

Reply:
When we do `List<TextUnitDTO>` to `AndroidStringDocument` we don't need the repository or targetLocale, we need it the other way around. But the class isn't stateful, each call to `AndroidStringDocumentMapper.mapToTextUnits` is self-contained.

Reply:
right, so it looks like a component would make more sense, and pass the required parameters to each function would make it more clearer what is needed.